cider/exec/nextgen/jitlib/base/JITTuple.h

```diff
@@ -64,6 +65,44 @@ class JITTuple final : public JITBaseValue {
   boost::container::small_vector<JITBaseValue*, DefaultVectorSize> children;
 };
 
+class JITExprValue {
+ public:
+  JITExprValue() = default;
+  ~JITExprValue() = default;
+  JITExprValue(const JITExprValue&) = delete;
+  JITExprValue(JITExprValue&&) = delete;
+
+  template <typename T>
+  JITExprValue(T&& ptrs, bool is_variadic = false)
+      : ptrs_(std::forward<T>(ptrs)), is_variadic_(is_variadic) {}
+
+  // convert JITValuePointer to JITExprValue
+  JITExprValue(JITValuePointer&& val) {
+    is_variadic_ = false;
+    ptrs_.push_back(std::move(val));
```

Comment:
```move``` is redundant because the ```val``` is a rvalue

Reply:
no, val is a lvalue. you can see this test. https://godbolt.org/z/snK48Mzoc